Xcode/SDL/pkg-support/resources/CMake/sdl3-config-version.cmake

```diff
@@ -3,12 +3,12 @@
 # SDL CMake version configuration file:
 # This file is meant to be placed in Resources/CMake of a SDL3 framework
 
-if(NOT EXISTS "${CMAKE_CURRENT_LIST_DIR}/../../Headers/SDL_version.h")
-    message(AUTHOR_WARNING "Could not find SDL_version.h. This script is meant to be placed in the Resources/CMake directory of SDL3.framework")
+if(NOT EXISTS "${CMAKE_CURRENT_LIST_DIR}/../../Headers/SDL3/SDL_version.h")
```

Comment:
I don't think this is right. For frameworks, the include syntax `<SDL3/SDL.h>` means "look in the SDL3 framework for the SDL.h header in the Headers folder". There isn't an SDL3 subdirectory in this case.

Reply:
I've mounted SDL2.framework, and am replacing SDL2 with SDL3.

These cmake files are stored in `SDL3.framework/Versions/A/Resources/CMake`.
So it will check the header `SDL3.framework/Versions/A/Headers/SDL3/SDL_version.h`.

Shouldn't there be a `SDL3` subfolder? Otherwise it is include incompatible with the other platforms.

Reply:
If the framework is in normal places, `<SDL3/SDL.h>` means SDL.h under `Headers` directory of `SDL3.framework`

Reply:
In other words: when users #include a header from a Framework, they already need to prepend the framework name to the include path (so SDL2 users already needed to #include <SDL2/SDL.h> when using the framework even though the files were directly in the Headers folder, unless they customized their include paths.)

Reply:
Oh, so the file `SDL3.framework/Headers/SDL_xxx.h` is only includeable as `#include <SDL3/SDL_xxx.h>`, not as `#include <SDL_xxx.h>`?

Reply:
Yes

Reply:
I think you are describing the side effects of passing `-F /some/path` and `-framework SDL3`.
We should make sure `SDL3_INCLUDE_DIRS` is also valid.

Somebody should check the following with the framework:
```cmake
cmake_minimum_required(VERSION 3.0)
project(sdl_framework_test C)

find_package(SDL3 REQUIRED)

file(WRITE source.c [[
#include <SDL3/SDL.h>

void some_function() {
  SDL_Init(SDL_INIT_EVERYTHING);
  SDL_Quit();
}
]])
add_library(some_object_library OBJECT source.c)
target_include_directories(some_object_library PRIVATE ${SDL3_INCLUDE_DIRS})
```

Reply:
How do I test this with the framework?

Reply:
`SDL3_INCLUDE_DIRS` is not compatible with the framework. Those variables should not be created with the the framework CMake support.

Reply:
I think if we really want them to be compatible both ways, we need to include the headers in the normal place for the framework, and include a separate copy of the headers in the Resources directory with the SDL3 path.

Reply:
In any case, your change here won't work with the framework layout, I think you can just omit this change from the commit.

Reply:
> How do I test this with the framework?

Configure the project with `-DCMAKE_PREFIX_PATH=/path/to/parent/folder/of/the/frameworks`.
`SDL3_INCLUDE_DIRS` is used by `sdl2-compat`, so should work.
Unless you are content with it not building on top of the framework.

Reply:
I reverted the changes to the sdl3 cmake config files for the framework.
But somebody able, please test `SDL3_INCLUDE_PATH`.